src/python/pants/backend/jvm/subsystems/zinc_language_mixin.py

```diff
@@ -22,6 +22,10 @@ def register_options(cls, register):
              fingerprint=True,
              help='The default for the "fatal_warnings" argument for targets of this language.')
 
+    register('--compiler-option-sets', advanced=True, default=[], type=list,
+             fingerprint=True,
+             help='The option sets to be enabled for the compilation of this target.')
```

Comment:
In this case, the `--compiler-option-sets` flag will end up defining the "default" options sets that are used for targets of a particular type (the `ZincLanguageMixin` is mixed in for "scala" and "java").

If a target passes the `compiler_option_sets=` argument, they'll be overriding the options sets.

Reply:
Awesome!
But 🤔 
So we don't want them to pass down the `compiler_option_sets=` argument?
Either way, we can't hardcode the default option sets just yet, can we? 

I'm not entirely sure where you were going with this comment 😅 

PS. I'm using a `list` instead of a `set` because `set`s are not json serializable!

Reply:
> So we don't want them to pass down the compiler_option_sets= argument?

Yes, they are allowed to do that. But if they do _not_ do that, we use the default from this option.

Reply:
This should be the same as with `fatal_warnings`... the option defined in `ZincLanguageMixin` is the default value of `fatal_warnings` for a target... if they don't set the option explicitly on their target, we use the default.

Reply:
The help for this option should likely be updated to align with the help for the deprecated `fatal_warnings` option: ie:
```
The default for the "compiler_option_sets" argument for targets of this language.
```